src/MooLite/plugins/WhisperManager/WhisperManagerPlugin.ts

```diff
@@ -0,0 +1,138 @@
+
+    _messages: ChatMessage[] = [];
+    _conversations: Record<string, Conversation> = {};
+    _activeConversationName: string = "";
+
+    config: PluginConfig[] = [
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide System Messages",
+            description: "Hide system messages from the conversation list",
+            key: "hide-system-messages",
+            value: false,
+        },
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide Mod Messages",
+            description: "Hide mod messages from the conversation list",
+            key: "hide-mod-messages",
+            value: false,
+        },
+    ];
+
+    tab: MooLiteTab = {
+        icon: "💬",
+        pluginName: this.name,
+        componentName: "WhisperManagerPluginDisplay",
+        component: markRaw(WhisperManagerPluginDisplay),
+        width: "w-80",
+    };
+
+    initialize(game: Game): void {
+        super.initialize(game);
+        console.log("WhisperManagerPlugin initialized");
+    }
+
+    public get hideSystemMessages(): boolean {
+        return this.getConfig("hide-system-messages").value;
+    }
+
+    public get hideModMessages(): boolean {
+        return this.getConfig("hide-mod-messages").value;
+    }
+
+    public get activeConversation(): Conversation {
+        return this._conversations[this._activeConversationName];
+    }
+
+    public get activeConversationName(): string {
+        return this._activeConversationName;
+    }
+
+    public get messages(): ChatMessage[] {
+        return this._messages;
+    }
+
+    public get conversations(): Record<string, Conversation> {
+        let conversations: Record<string, Conversation> = this.hideSystemMessages
+            ? (({ System, ...o }) => o)(this._conversations)
+            : this._conversations;
+        conversations = this.hideModMessages ? (({ Mods, ...o }) => o)(conversations) : conversations;
+
+        return conversations;
+    }
+
+    public get game(): Game {
+        return this._game;
+    }
+
+    public populateDevConversations(): void {
+        Object.entries(conversationData).forEach(([name, messages]) => {
+            this._conversations[name] = new Conversation(messages);
+        });
+    }
+
+    public populateConversations(): void {
+        if (Object.keys(this._conversations).length > 0) return;
+        for (const message of this._game.chatWhisperHistory) {
+            this._addToConversations(message);
+        }
+    }
+
+    public setActiveConversation(name: string): void {
+        if (!this._conversations[name]) return;
+        this._activeConversationName = name;
+        if (this._conversations[name].unread) {
+            this._conversations[name].toggleUnread();
+        }
+    }
+
+    onChatMessage(message: ChatMessage): void {
+        const { channel, isModMessage, isSystemMessage } = message;
+        const theseAreTheDroidsWereLookingFor =
```

Comment:
Funny, but please rename to something sensible 😄

Reply:
Aw c'mon, let me keep this one :D I would just rename it to something like `shouldHandleMessage`. I'll change if you insist but I like it enough to leave this comment. Worth a shot.